src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: string): GroupsMap<T> {
+  let arr: string[] = [];
+  const result: GroupsMap<T> = {};
+
+  arr = items.map((item) => item[key]);
```

Comment:
Instead of using a separate array `arr`, you can directly create the unique keys in the resulting object.